src/main/java/br/unb/cic/analysis/df/OverridingAssignmentAnalysis.java

```diff
@@ -31,62 +30,71 @@ public OverridingAssignmentAnalysis(Body methodBody, AbstractMergeConflictDefini
         super(methodBody, definition);
     }
 
+    @Override
+    protected void flowThrough(FlowSet<DataFlowAbstraction> in, Unit u, FlowSet<DataFlowAbstraction> out) {
+        detectConflict(in, u);
+        FlowSet<DataFlowAbstraction> temp = new ArraySparseSet<>();
+
+        FlowSet<DataFlowAbstraction> killSet = new ArraySparseSet<>();
+
+        for (DataFlowAbstraction item : in) {
+            if (statementEquals(item, u)) {
+                killSet.add(item);
+            }
+        }
+        in.difference(killSet, temp);
+        temp.union(gen(u, in), out);
+    }
+
     @Override
     protected FlowSet<DataFlowAbstraction> gen(Unit u, FlowSet<DataFlowAbstraction> in) {
         FlowSet<DataFlowAbstraction> res = new ArraySparseSet<>();
         if (isSourceStatement(u) || isSinkStatement(u)) {
             u.getUseAndDefBoxes().stream().filter(v -> v.getValue() instanceof Local).forEach(v -> {
-                Statement stmt = isSourceStatement(u) ? findSourceStatement(u) : findSinkStatement(u);
+                Statement stmt = isSourceStatement(u)
+                        ? findSourceStatement(u)
+                        : findSinkStatement(u);
                 res.add(new DataFlowAbstraction((Local) v.getValue(), stmt));
             });
-        } else if (u.getDefBoxes().size() > 0) {
-
-            u.getDefBoxes().stream().filter(v -> v.getValue() instanceof Local).forEach(v -> {
-                String localName = getLocalName((Local) v.getValue());
-
-                for (DataFlowAbstraction defsIn: in){
-                    String defInName = getLocalName(defsIn.getLocal());
-                    //if u not in IN, then add it
-                    if (!defInName.equals(localName)){
-                        res.add(new DataFlowAbstraction(defsIn.getLocal(), defsIn.getStmt())); //update an element in IN
-                        break; //Do not necessary check others elements
-                    }
-                }
-            });
         }
         return res;
     }
 
-
     @Override
-    protected void detectConflict(FlowSet<DataFlowAbstraction> in, Unit u){
-        if (!(isSinkStatement(u) || isSourceStatement(u))){
-            return;
-        }
+    protected void detectConflict(FlowSet<DataFlowAbstraction> in, Unit u) {
         List<DataFlowAbstraction> left = new ArrayList<>();
         List<DataFlowAbstraction> right = new ArrayList<>();
 
+        if (!(isSinkStatement(u) || isSourceStatement(u))) {
```

Comment:
This condition could go come before of the declaration list, because if it is true, it is not need do it.

Reply:
done!